### ISSUE TITLE
Unable to Precache all visited pages, just home page

### ISSUE THREAD

**Issue by andrewbiang888** [nuxt-community/pwa-module#97]
I've been messing with the Workbox feature for a few days and can't seem to get the precache option to work as I expect it to (probably user error). I made a restaurant review app for a Udacity class and could not get restaurant page to cache. The restaurant url would look like `http://localhost:3333/restaurant?id=<number>`. Not sure if that is what is causing my issue.

**What I want to happen**:
Load homepage (initial load), click on a restaurant page, disconnect form the internet, reload the page and see the same content.

**What happens**:
Load homepage (initial load), click on a restaurant page, disconnect form the internet, reload the page and get "This site can’t be reached" message.

**What I've tried**:

- Adding `cachingExtensions: 'http://localhost:3333/restaurant'` to workbox object in `nuxt.config.js`
- Adding this to `runtimeCaching` array: 
```
     {
        urlPattern: new RegExp('^http://localhost:1337/restaurant'),
        handler: 'cacheFirst',
        strategyOptions: {
          cacheableResponse: {
            statuses: [0, 200]
          }
        }
      }
```

**Where I'm at now**:

Here is my workbox object in `nuxt.config.js`.

```
workbox: {
    // importScripts: [
    //     'test-sw.js'
    // ],
    ignoreUrlParametersMatching: [/./],
    skipWaiting: true,
    cachingExtensions: 'http://localhost:3333/restaurant',
    runtimeCaching: [
      {
        // To match cross-origin requests, use a RegExp that matches
        // the start of the origin:
        urlPattern: new RegExp('^http://localhost:1337/.*'),
        handler: 'cacheFirst',
        strategyOptions: {
          cacheableResponse: {
            statuses: [0, 200]
          }
        }
      },
      {
        urlPattern: new RegExp('^https:\/\/api\.tiles\.mapbox\.com\/.*'),
        handler: 'cacheFirst',
        method: 'GET',
        strategyOptions: {
          cacheableResponse: {
            statuses: [0, 200]
          }
        }
      },
      {
        urlPattern: new RegExp('^http://localhost:3333/img/.*'),
        handler: 'cacheFirst',
        method: 'GET',
        strategyOptions: {
          cacheName: 'images',
          cacheableResponse: { statuses: [0, 200, 304] }
        }
      }
    ]
  }
```

Am I misunderstanding how this should work or is there a way to make it happen? 

<div align="right"><sub><em>This question is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/pwa-module/issues/c73">#c73</a>)</em></sub></div>

**Comment by hanjeahwan** [comment]
i facing this issue too, the new version is not caching the generated html page when offline and reload is not load from the service worker, after i check the sw.js file, the 'workbox.precaching.precacheAndRoute' array is not include generated folder and .html.
if anyone know how to solve this much appreciated

**Comment by galvez** [comment]
@hanjeahwan can you share your exact configuration?

**Comment by pi0** [comment]
Should be fixed with #100.

**Comment by agcty** [comment]
I'm having a similar problem, also with restaurant pages 😅. When using issuing npm run build && npm run start on my local machine I can access restaurant pages such aus localhost:3000/restaurant/uuid and also reload them without any problems. However, when deploying I cannot reload these pages and instead get a "This site can't be reached error" from Chrome.

I tried with and without a custom config. The thing I tried is:

```javascript
runtimeCaching: [
      {
        urlPattern: new RegExp("/restaurant/(.*)"),
        handler: "networkFirst"
      }
    ]
```

However, without using a custom config I imagined the offline flag turned on by default should handle such cases. Also from the documentation I can't understand what "pagesURLPattern" is used for.
